lib/matplotlib/tests/test_subplots.py

```diff
@@ -212,3 +212,37 @@ def test_dont_mutate_kwargs():
 def test_subplot_factory_reapplication():
     assert maxes.subplot_class_factory(maxes.Axes) is maxes.Subplot
     assert maxes.subplot_class_factory(maxes.Subplot) is maxes.Subplot
+
+
+@pytest.mark.parametrize("width_ratios", [None, [1, 3, 2]])
+@pytest.mark.parametrize("height_ratios", [None, [1, 2]])
+@check_figures_equal(extensions=['png'])
+def test_width_and_height_ratios(fig_test, fig_ref,
+                                 height_ratios, width_ratios):
+    fig_test.subplots(2, 3, height_ratios=height_ratios,
+                      width_ratios=width_ratios)
+    fig_ref.subplots(2, 3, gridspec_kw={
+                     'height_ratios': height_ratios,
+                     'width_ratios': width_ratios})
+
+
+@pytest.mark.parametrize("width_ratios", [None, [1, 3, 2]])
+@pytest.mark.parametrize("height_ratios", [None, [1, 2]])
```

Comment:
I think this needs to be [[1,2], None] ?

Reply:
I think this does all four permutations doesn't it?

Reply:
Yes, sorry, forgot how parameterization works !